includes/general.php

```diff
@@ -1090,7 +1091,14 @@ function pods_shortcode_run( $tags, $content = null ) {
 		echo $pod->pagination( $tags['pagination_label'] );
 	}
 
-	echo $pod->template( $tags['template'], $content );
+	$content = $pod->template( $tags['template'], $content );
+	$content = trim( $content );
+
+	if ( empty( $content ) && ! empty( $tags['not_found'] ) ) {
+		$content = $pod->do_magic_tags( $tags['not_found'] );
+	}
+
+	echo $content;
```

Comment:
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$content'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$content'.

<sub>hash: 1100ff42becac100619afdca4cbf44de</sub>